packages/manager/src/features/CloudView/Dashboard/CloudPulseDashboardWithFilters.tsx

```diff
@@ -0,0 +1,186 @@
+import { Grid, styled } from '@mui/material';
+import React from 'react';
+
+import CloudViewIcon from 'src/assets/icons/Monitor.svg';
+import { CircleProgress } from 'src/components/CircleProgress';
+import { ErrorState } from 'src/components/ErrorState/ErrorState';
+import { Paper } from 'src/components/Paper';
+import { Placeholder } from 'src/components/Placeholder/Placeholder';
+import { useCloudViewDashboardByIdQuery } from 'src/queries/cloudview/dashboards';
+
+import { CloudPulseDashboardFilterBuilder } from '../Overview/DashboardFilterBuilder';
+import { FILTER_CONFIG } from '../Utils/FilterConfig';
+import { CloudPulseWidgetFilters } from '../Widget/CloudViewWidget';
+import { CloudPulseDashboard, DashboardProperties } from './Dashboard';
+
+export interface CloudPulseDashboardWithFiltersProp {
+  dashboardId: number;
+  resource: number; // the id of the resources
+}
+
+export const CloudPulseDashboardWithFilters = React.memo(
+  (props: CloudPulseDashboardWithFiltersProp) => {
+    const { data: dashboard, isError } = useCloudViewDashboardByIdQuery(
+      props.dashboardId,
+      undefined
+    );
+
+    const [filterValue, setFilterValue] = React.useState<{
+      [key: string]: any;
+    }>({});
+
+    const filterReference: { [key: string]: any } = React.useRef({});
+
+    const emitFilterChange = (filterKey: string, value: any) => {
+      filterReference.current[filterKey] = value;
+      setFilterValue({ ...filterReference.current });
+    };
+
+    const getDashboardProperties = (): DashboardProperties => {
+      const dashboardProps: DashboardProperties = {
```

Comment:
```suggestion
      return {
        dashboardId: props.dashboardId,
        duration: undefined,
        resources: [],
        globalFilters: constructDimensionFilters(),
        savePref: false
      }
```

we can directly construct & return